ci/dev/vscode.patch

```diff
@@ -3797,18 +3797,25 @@ index 85d83f37da179a1e39266cf72a02e971f590308e..0659738b36df1747c9afcabf8d9abf26
  		};
  
 diff --git a/src/vs/workbench/services/environment/browser/environmentService.ts b/src/vs/workbench/services/environment/browser/environmentService.ts
-index a8d43045ecc8cbe04b3f8440cff16d42aadbcad0..cd589c6f75eccbeefbf364d426ac882396b26fb4 100644
+index a8d43045ecc8cbe04b3f8440cff16d42aadbcad0..8e122c761ac7ddfee11f9dda2ac5e845b893cc28 100644
 --- a/src/vs/workbench/services/environment/browser/environmentService.ts
 +++ b/src/vs/workbench/services/environment/browser/environmentService.ts
-@@ -119,8 +119,18 @@ export class BrowserWorkbenchEnvironmentService implements IWorkbenchEnvironment
+@@ -119,8 +119,25 @@ export class BrowserWorkbenchEnvironmentService implements IWorkbenchEnvironment
  	@memoize
  	get logFile(): URI { return joinPath(this.options.logsPath, 'window.log'); }
  
-+	// NOTE@coder: Use the regular path for extensions that write directly to disk
-+	// instead of using the VS Code API.
++	// NOTE@coder: Use the same path in // ../../../../platform/environment/node/environmentService.ts
++	// and don't use the user data scheme. This solves two problems:
++	//  1. Extensions running in the browser (like Vim) might use these paths
++	//     directly instead of using the file service and most likely can't write
```

Comment:
What do you mean by

> most likely can't write to `/User` on disk.

They can't at all right since they're running in the browser.

Reply:
They can since client-side extensions run in our shimmed Node environment which calls out to the server for things like `fs.write` etc.

Reply:
Oh gotchu.